### ISSUE TITLE
[Tune] TrialCheckpoint PLACEHOLDER fails DNS resolution

### ISSUE THREAD

**Issue by max0x7ba** [ray-project/ray#26047]
### What happened + What you expected to happen

I use `ExperimentAnalysis` to load tuning results in Google Compute Engine machines with no internet connection.

This ends up calling https://github.com/ray-project/ray/blame/26ae3a02399fb4c7af243dc9e9883d35d2034882/python/ray/tune/cloud.py#L85
```
        # Checkpoint does not allow empty data, but TrialCheckpoint
        # did. To keep backwards compatibility, we use a placeholder URI
        # here, and manually set self._uri and self._local_dir later.
        PLACEHOLDER = "s3://placeholder"
        Checkpoint.__init__(self, uri=PLACEHOLDER)
```
Which fails with:

```
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/max/src/qr/es/show_tune_results.py", line 156, in <module>
    main()
  File "/home/max/src/qr/es/show_tune_results.py", line 125, in main
    df, tune_mm = load_best_results1(options, symbol, True)
  File "/home/max/src/qr/es/tune_results.py", line 72, in load_best_results1
    df['model_dir'] = df['logdir'].map(get_best_checkpoint)
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/pandas/core/series.py", line 4237, in map
    new_values = self._map_values(arg, na_action=na_action)
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/pandas/core/base.py", line 880, in _map_values
    new_values = map_f(values, mapper)
  File "pandas/_libs/lib.pyx", line 2870, in pandas._libs.lib.map_infer
  File "/home/max/src/qr/es/tune_results.py", line 65, in get_best_checkpoint
    checkpoint = a.get_best_checkpoint(d)
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/ray/tune/analysis/experiment_analysis.py", line 469, in get_best_checkpoint
    return TrialCheckpoint(local_path=best_path, cloud_path=cloud_path)
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/ray/tune/cloud.py", line 86, in __init__
    Checkpoint.__init__(self, uri=PLACEHOLDER)
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/ray/ml/checkpoint.py", line 131, in __init__
    local_path = _get_local_path(uri)
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/ray/ml/checkpoint.py", line 457, in _get_local_path
    if path is None or is_non_local_path_uri(path):
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/ray/ml/utils/remote_storage.py", line 74, in is_non_local_path_uri
    if bool(get_fs_and_path(uri)[0]):
  File "/home/max/anaconda3/envs/torch2/lib/python3.8/site-packages/ray/ml/utils/remote_storage.py", line 104, in get_fs_and_path
    fs, path = pyarrow.fs.FileSystem.from_uri(uri)
  File "pyarrow/_fs.pyx", line 350, in pyarrow._fs.FileSystem.from_uri
  File "pyarrow/error.pxi", line 144, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 115, in pyarrow.lib.check_status
OSError: When resolving region for bucket 'placeholder': AWS Error [code 99]: curlCode: 28, Timeout was reached
```

Can you remove this unnecessary DNS resolution of "s3://placeholder", please? 

@krfricke 

### Versions / Dependencies

ray-1.13.0

### Reproduction script

No reproduction script is available.

### Issue Severity

High: It blocks me from completing my task.

**Comment by krfricke** [comment]
There is an open draft PR removing TrialCheckpoint completely: https://github.com/ray-project/ray/pull/25406
I'll likely work on this in the next 2 weeks

**Comment by krfricke** [comment]
This should have been resolved by #25406
